### ISSUE TITLE
King Danger

### ISSUE THREAD

**Issue by crespire** [crespire/ruby_chess#1]
Currently, we only check for King dangers, which is a ray based query ignoring blocks.

How do we account for cells in `danger_zone` that are not under immediate attack?

Sometimes these are legal moves but sometimes they are not.

**Comment by crespire** [comment]
Build attacks map as well. Case of enemy pawn: use captures otherwise, use moves.


danger_zone - attacks = cells we must test to figure out if they're legal moves.

```
for each cell
# deep-copy the current game
move_legal?
if not legal, remove from pseudo.


def move_legal?(game, origin, tested_destination)
  @game.update_loc
  attackers, _ = @game.movement.get_enemies
  attackers.empty?
end
```

Something like that?

**Comment by crespire** [comment]
This issue is resolved in the meantime with c5a1f5789174bee3ae77c8913efaee03f620b11c
